packages/flutter_test/lib/src/finders.dart

```diff
@@ -327,6 +335,32 @@ class _LastFinder extends Finder {
   }
 }
 
+class _HitTestableFinder extends Finder {
+  _HitTestableFinder(this.parent, this.offset);
+
+  final Finder parent;
+  final FractionalOffset offset;
+
+  @override
+  String get description => '${parent.description} (considering only hit-testable ones)';
+
+  @override
+  Iterable<Element> apply(Iterable<Element> candidates) sync* {
+    for (final Element candidate in parent.apply(candidates)) {
+      final RenderBox box = candidate.renderObject;
+      assert(box != null);
+      final Offset absoluteOffset = box.localToGlobal(offset.alongSize(box.size));
+      final HitTestResult hitResult = new HitTestResult();
+      WidgetsBinding.instance.hitTest(hitResult, absoluteOffset);
+      for (final HitTestEntry entry in hitResult.path) {
+        if (entry.target == candidate.renderObject) {
```

Comment:
nit: no braces on the inner if

Also, presumably you should break out of the inner loop if you yield?

Reply:
> nit: no braces on the inner if

Done.

> Also, presumably you should break out of the inner loop if you yield?

The intention here is to find all hit-testable elements. Then it's up to the test to decide what to do with them. If a test only cares about the first element, it can be achieved by `finder.hitTestable().first`.

Reply:
Oh wait, the _inner_ loop! Yes, you are right. Done.